client/src/pages/SignUp.js

```diff
@@ -0,0 +1,243 @@
+        <div className={classes.leftSideContainer}>
+          <div className={classes.titleLogo}>
+            <Typography variant="h4">CHEF'S MENU</Typography>
+          </div>
+          <Card className={classes.card}>
+            <CardContent>
+              <Typography className={classes.title} gutterBottom>
+                Create an account
+              </Typography>
+              <form autoComplete="off">
+                <Box display="flex">
+                  <TextField
+                    className={classes.textField}
+                    required
+                    autoFocus
+                    id="signupName"
+                    name="signupName"
+                    label="Name"
+                    placeholder="Enter your name"
+                    error={this.state.usernameError !== "" ? true : false}
+                    variant="outlined"
+                    onChange={this.validateChange}
+                    helperText={this.state.usernameError}
+                  />
+                </Box>
+                <Box display="flex">
+                  <TextField
+                    className={classes.textField}
+                    required
+                    name="signupEmail"
+                    label="Email"
+                    placeholder="Enter your e-mail address"
+                    variant="outlined"
+                    error={this.state.emailError !== "" ? true : false}
+                    helperText={this.state.emailError}
+                    onChange={this.validateChange}
+                  />
+                </Box>
+                <Box display="flex">
+                  <TextField
+                    className={classes.textField}
+                    required
+                    type="password"
+                    name="signupPwd"
+                    label="Password"
+                    error={this.state.passwordError !== "" ? true : false}
+                    placeholder="Enter password"
+                    variant="outlined"
+                    helperText="Minimum 6 characters"
+                    onChange={this.validateChange}
+                  />
+                </Box>
+                <Box display="flex">
+                  <TextField
+                    className={classes.textField}
+                    required
+                    type="password"
+                    name="signupPwdConfirm"
+                    label="Confirm Password"
+                    placeholder="Re-enter password"
+                    variant="outlined"
+                    onChange={this.validateChange}
+                    error={
+                      this.state.passwordConfirmError === "Mismatch"
+                        ? true
+                        : false
+                    }
+                    helperText={
+                      this.state.passwordConfirmError === "Mismatch"
+                        ? "Passwords not matching"
+                        : ""
```

Comment:
In all the above cases you were using the actual error text to display. Makes sense to do that here

Reply:
Got it. Done.